### PR TITLE
Automatically build bundles every 2 weeks

### PR DESCRIPTION
**Pull request by andresmascl**
Fixes [Regularly build and publish airgapped bundles with airgapped Trivy image](https://d2iq.atlassian.net/browse/D2IQ-93932)

[D2IQ-93932]: https://d2iq.atlassian.net/browse/D2IQ-93932?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

The build is triggered by a scheduled (cronjob )Github Actions workflow.

The first step in the workflow clones the dkp-insights repository, and checks out the last 3 release tags.  It then gathers the image names of the trivy bundles in all 3 of these releases.

In the second step it re-builds the trivy-bundles image (with an updated CVE database) and pushes it to Docker Hub with the same image name and tag used in each of the dkp-insights releases.

The workflow was tested by triggering it manually.  But it has not been tested by triggering it using the cron job.  According to [this blog post](https://upptime.js.org/blog/2021/01/22/github-actions-schedule-not-working/) it can take several hours for the cron job to be triggered for the first time, and also that there is no guarantee that Github will trigger this jobs timely at the hours they are scheudled.

According to this [blog postblog](https://devopsjournal.io/blog/2022/08/12/workflows-not-starting) post scheduled workflows will not be triggered in branches other than the default branch due to security reasons.  For this the current PR needs to be merged into main for the cron job to run.

Here are 2 sample PRs opened by this workflow, for branches [main](https://github.com/mesosphere/dkp-insights/pull/596), and for [release-v0.3](https://github.com/mesosphere/dkp-insights/pull/595)

### CODE REVIEW

File: .github/workflows/rebuild-trivy-bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+        run: |
+          wget "https://go.dev/dl/$(curl 'https://go.dev/VERSION?m=text').linux-amd64.tar.gz"
+          sudo tar -C /usr/local -xzf go*.linux-amd64.tar.gz
+
+      # - name: Create folder for cloning dkp-insights
+      #   if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+      #   run:
+      #     sudo mkdir dkp-insights
+      #     sudo chmod  0777 ./dkp-insights
```
Comment (kaiwalyajoshi) [inline .github/workflows/rebuild-trivy-bundle.yaml]: Let's delete these if we're not using them?

File: .github/workflows/rebuild-trivy-bundle.yaml [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+  update-dkp-insights-release-branches:
+    needs: [rebuild-trivy-bundles-images, determine-if-workflow-should-run]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Reinstall Go for Current User
+        if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+        run: |
+          wget "https://go.dev/dl/$(curl 'https://go.dev/VERSION?m=text').linux-amd64.tar.gz"
+          sudo tar -C /usr/local -xzf go*.linux-amd64.tar.gz
```
Comment (kaiwalyajoshi) [inline .github/workflows/rebuild-trivy-bundle.yaml]: We don't actually end up building `dkp-insights` here (we did previously). 

Would it be possible to remove steps the build steps, involving use of  `working-directory: //usr/local/go/src/github.com/mesosphere/dkp-insights` to simplify this PR?

This workflow doesn't run too frequently or in a critical path so runtime isn't as much of a concern, but removing the need to be in `GOPATH` etc... should simplify this PR.

File: .github/workflows/rebuild-trivy-bundle.yaml [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+      #   if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+      #   run:
+      #     sudo mkdir dkp-insights
+      #     sudo chmod  0777 ./dkp-insights
+      
+      - name: Create folder for moving dkp-insights within go path
+        if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+        run: |
+          sudo mkdir -p //usr/local/go/src/github.com/mesosphere/dkp-insights
+          sudo chmod  -R  0777 //usr/local/go/src/github.com/mesosphere/dkp-insights
```
Comment (kaiwalyajoshi) [inline .github/workflows/rebuild-trivy-bundle.yaml]: (same as previous comment, we don't build `dkp-insights` anymore)

File: .github/workflows/rebuild-trivy-bundle.yaml [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+          submodules: 'true'
+          path: dkp-insights
+          token: ${{ secrets.MERGEBOT_TOKEN }}
+
+      - name: Move dkp-insights Repository to GOPATH
+        if: needs.determine-if-workflow-should-run.outputs.WORKFLOW_SHOULD_RUN
+        run: |
+          mv ./dkp-insights/* ./dkp-insights/.[!.]* ./dkp-insights/..?* //usr/local/go/src/github.com/mesosphere/dkp-insights/ || :
+          cd //usr/local/go/src/github.com/mesosphere/dkp-insights
+          ls -a
```
Comment (kaiwalyajoshi) [inline .github/workflows/rebuild-trivy-bundle.yaml]: (same as previous comment, we don't build `dkp-insights` anymore)

File: .github/workflows/rebuild-trivy-bundle.yaml [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        working-directory: //usr/local/go/src/github.com/mesosphere/dkp-insights
+        env:
+          GITHUB_TOKEN: ${{ secrets.MERGEBOT_TOKEN }}
+        run: |
+          echo "setup environment"
+          export PATH=$PATH:/usr/local/go/bin
+          echo "GOPATH=/usr/local/go" >> ~/.bashrc
+          export GOPATH=/usr/local/go
+          export IR=${GOPATH}/src/github.com/mesosphere/dkp-insights
+          export PATH=${IR}/.local/tools:${PATH}
```
Comment (kaiwalyajoshi) [inline .github/workflows/rebuild-trivy-bundle.yaml]: (same as previous comment, we don't build `dkp-insights` anymore)

File: .github/workflows/rebuild-trivy-bundle.yaml (hunk 1 above)
Comment (andresmascl) [reply .github/workflows/rebuild-trivy-bundle.yaml]: Ok done!

**Review by kaiwalyajoshi — APPROVED** [review]
Overall PR looks good and does exactly what we want.
In previous iterations, we did a build of `dkp-insights` which required installation of Golang and it had to be under `GOPATH` etc.... This is not needed anymore and the workflow code can be simplified to remove these parts which are effectively dead-code.

File: .github/workflows/rebuild-trivy-bundle.yaml (hunk 5 above)
Comment (andresmascl) [reply .github/workflows/rebuild-trivy-bundle.yaml]: You're right.  I'll correct them all now

**Comment by andresmascl** [comment]
> Overall PR looks good and does exactly what we want. In previous iterations, we did a build of `dkp-insights` which required installation of Golang and it had to be under `GOPATH` etc.... This is not needed anymore and the workflow code can be simplified to remove these parts which are effectively dead-code.

Yeah, ok.  I think I solved all the unnecesary steps now and the commit was tested 👌

**Pull request merged**